### ISSUE TITLE
sam-beta-cdk fails to CDK synth lambda functions built from Docker images

### ISSUE THREAD

**Issue by crawfobw** [aws/aws-sam-cli#3129]
### Description:
CDK synth fails for lambda functions that build their code from ECR image code with 

### Steps to reproduce:
Using the hello-world SAM template in python with cdk. The following stack fails to synth via `sam-beta-cdk local`

```
from aws_cdk import core
from aws_cdk import aws_lambda as _lambda
from aws_cdk import aws_apigateway as _apigateway
from aws_cdk import aws_lambda_event_sources as _lambda_event_sources

# For consistency with other languages, `cdk` is the preferred import name for
# the CDK's core module.  The following line also imports it as `core` for use
# with examples from the CDK Developer's Guide, which are in the process of
# being updated to use `cdk`.  You may delete this import if you don't need it.


class AwsSamCliCdkHelloWorldStack(core.Stack):
    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self.lambda_event = _lambda_event_sources.ApiEventSource(
            method="get",
            path="/hello"
        )

        cdk_lambda = _lambda.Function(
            scope=self,
            id="cdk-hello-world-lambda",
            runtime=_lambda.Runtime.FROM_IMAGE,
            function_name="cdk-hello-world-lambda-function",
            description="Lambda function deployed using AWS CDK Python",
            code=_lambda.EcrImageCode.from_asset_image(
                directory="./hello_world",
                cmd=["app.lambda_handler"]
            ),
            handler=_lambda.Handler.FROM_IMAGE,
            events=[self.lambda_event]
        )

        api = _apigateway.LambdaRestApi(
            scope=self,
            id="cdk-hello-world-rest-api",
            handler=cdk_lambda,
            proxy=False
        )

        hello_world = api.root.add_resource("hello")
        hello_world.add_method('GET')

```

### Observed result:
```
$ sam-beta-cdk local invoke sam-test/cdk-hello-world-lambda --debug
2021-08-01 19:32:29,053 | IAC Plugin getting project...
2021-08-01 19:32:29,055 | CDK Toolkit found at /opt/homebrew/bin/cdk
2021-08-01 19:32:29,055 | Synthesizing CDK App
2021-08-01 19:32:29,055 | command: ['/opt/homebrew/bin/cdk', 'synth', '--no-staging', '-o', '.aws-sam/.cdk-out']
2021-08-01 19:32:30,595 | Cloud assembly synthed at .aws-sam/.cdk-out
2021-08-01 19:32:30,595 | _validate_cloud_assembly: .aws-sam/.cdk-out
2021-08-01 19:32:30,597 | Found assets: {'/Users/crawfobw1/Documents/workspace/3meow/meow/sam-image/hello_world': <samcli.lib.iac.interface.ImageAsset object at 0x106820520>}
2021-08-01 19:32:30,597 | ca-stack_name: sam-image
2021-08-01 19:32:30,597 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2021-08-01 19:32:30,597 | Using config file: samconfig.toml, config environment: default
2021-08-01 19:32:30,597 | Expand command line arguments to:
2021-08-01 19:32:30,597 | --project_type=CDK --template_file=/Users/crawfobw1/Documents/workspace/3meow/meow/sam-image/template.yml --function_logical_id=sam-test/cdk-hello-world-lambda --no_event --layer_cache_basedir=/Users/crawfobw1/.aws-sam/layers-pkg --iac=<samcli.lib.iac.cdk.plugin.CdkPlugin object at 0x1026dfc40> --project=<samcli.lib.iac.interface.Project object at 0x1026f3a60> 
2021-08-01 19:32:30,597 | local invoke command is called
2021-08-01 19:32:30,597 | No Parameters detected in the template
2021-08-01 19:32:30,615 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': 'e8112f10-2644-4b19-8755-9fcf0fccfc56', 'installationId': '32892da2-489c-44ad-bb26-256326711294', 'sessionId': '4170973f-362d-4939-b80a-83f9fe3a0251', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.8.11', 'samcliVersion': '1.22.0.dev202107140310', 'metricSpecificAttributes': {'projectType': 'CDK'}, 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam-beta-cdk local invoke', 'duration': 18, 'exitReason': 'InvalidIntrinsicException', 'exitCode': 255}}]}
2021-08-01 19:32:31,181 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)
Traceback (most recent call last):
  File "/opt/homebrew/bin/sam-beta-cdk", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/iac/utils/helpers.py", line 55, in wrapper
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/commands/_utils/iac_validations.py", line 80, in wrapped
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 174, in wrapped
    raise exception  # pylint: disable=raising-bad-type
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 141, in wrapped
    return_value = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/cli/main.py", line 90, in wrapper
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/commands/local/invoke/cli.py", line 93, in cli
    do_cli(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/commands/local/invoke/cli.py", line 164, in do_cli
    with InvokeContext(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/commands/local/cli_common/invoke_context.py", line 188, in __enter__
    self._stacks = self._get_stacks()
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/commands/local/cli_common/invoke_context.py", line 398, in _get_stacks
    stacks, _ = SamLocalStackProvider.get_stacks(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_stack_provider.py", line 257, in get_stacks
    current = SamLocalStackProvider(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_stack_provider.py", line 55, in __init__
    self._template_dict: IacStack = self.get_template(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/providers/sam_base_provider.py", line 181, in get_template
    template_dict = resolver.resolve_template(ignore_errors=True)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py", line 253, in resolve_template
    self.resolve_resources_assets(ignore_errors)
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py", line 268, in resolve_resources_assets
    asset.source_local_image = self.intrinsic_property_resolver(
  File "/opt/homebrew/Cellar/aws-sam-cli-beta-cdk/202107140310/libexec/lib/python3.8/site-packages/samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py", line 196, in intrinsic_property_resolver
    raise InvalidIntrinsicException("Missing Intrinsic property in {}".format(parent_function))
samcli.lib.intrinsic_resolver.invalid_intrinsic_exception.InvalidIntrinsicException: Missing Intrinsic property in Code
```

### Expected result:
the application to synth correctly. (Note `$ cdk synth` completes and produces a CFT)

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: OSx 11.4
2. `sam --version`: SAM CLI, version 1.27.2
3. AWS region: us-east-2
4. `$ sam-beta-cdk --version
SAM CLI, version 1.22.0.dev202107140310`

**Comment by jfuss** [comment]
@crawfobw Thanks for the report. Talking with the team working on the CDK support, this seems like a bug we introduced in the latest beta. We were already looking into this and figuring out a patch. 

Updating labels

**Comment by mildaniel** [comment]
A fix has been released in [v1.29.0.dev202108311500](https://github.com/aws/aws-sam-cli/releases/tag/v1.29.0.dev202108311500). Closing the issue.
